src/crypto-providers/ledgerCryptoProvider.ts

```diff
@@ -92,13 +107,33 @@ export const LedgerCryptoProvider: () => Promise<CryptoProvider> = async () => {
     outputIndex: input.outputIndex,
   })
 
+  const prepareTokenBundle = (
+    multiAssets: _MultiAsset[],
+  ): LedgerAssetGroup[] => {
+    if (multiAssets.length > 0 && !isFeatureSupportedForVersion(LedgerCryptoProviderFeature.MULTI_ASSET)) {
+      throw Error(Errors.LedgerMultiAssetsNotSupported)
+    }
+    return multiAssets.map(({ policyId, assets }) => {
+      const tokens = assets.map(({ assetName, coins }) => ({
+        assetNameHex: assetName.toString('hex'),
+        amountStr: coins.toString(),
+      }))
+      return {
+        policyIdHex: policyId.toString('hex'),
+        tokens,
+      }
+    })
+  }
+
   const prepareChangeOutput = (
-    coins: BigInt,
+    amountStr: string,
```

Comment:
what is the reason behind switching fromBigInt to string?

I would advocate for keeping the "native" format of things up to the very moment they are meant to leave the system and need to be translated to minimize/avoid mixing an ambiguous representation of the same thing in the codebase which only makes it more susceptible to bugs.

Reply:
Reverted